lib/menu.rb

```diff
@@ -0,0 +1,23 @@
+class Menu
+  attr_reader :items
+
+  def initialize
+    @items = {}
+  end
+
```

Comment:
Since the menu key is the same as the item id, you could do this

```
def add(item)
  @items[item[:id]] = item
end
```

Though it feels a bit duplicative to have the key being the same as the ID.  I guess that as I read more of your code, I will understand why you chose to do this! :)